cache/engine/client/client.go

```diff
@@ -31,17 +32,18 @@ var ctx = context.Background()
 
 // RedisClient - Redis Client structure.
 type RedisClient struct {
-	*goredislib.Client
-	*redsync.Redsync
-	Name   string
-	Mutex  map[string]*redsync.Mutex
-	logger *log.Logger
+	Client  goredislib.UniversalClient
```

Comment:
what's the reason for not keeping this as a pointer anymore?

Reply:
Because UniversalClient doesn't implement *"github.com/go-redis/redis/v8".UniversalClient. An UniversalClient instance represents either a pointer to a ClusterClient, a FailoverClient, or a single-node Client:
_ UniversalClient = (*Client)(nil)
_ UniversalClient = (*ClusterClient)(nil)
_ UniversalClient = (*Ring)(nil)
https://github.com/redis/go-redis/blob/master/universal.go#L225

So, it ends up becoming a pointer (line 45).
![image](https://github.com/fabiocicerchia/go-proxy-cache/assets/76789260/2343a553-58b0-48f8-86ac-879ac59cba3f)